src/main/java/nextstep/ladder/domain/ladder/Ladder.java

```diff
@@ -0,0 +1,45 @@
+package nextstep.ladder.domain.ladder;
+
+import nextstep.ladder.domain.generator.RandomBooleanGenerator;
+import nextstep.ladder.domain.user.Position;
+
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+
+public class Ladder {
+
+    private final List<Row> rows;
+
+    public Ladder(List<Row> rows) {
+        this.rows = rows;
+    }
+
+    public static Ladder create(int rowSize, int pointSize) {
+        return new Ladder(
+                IntStream.range(0, rowSize)
+                        .mapToObj(index -> Row.initialize(pointSize, new RandomBooleanGenerator()))
+                        .collect(Collectors.toUnmodifiableList()));
+    }
+
+    public List<Row> getRows() {
+        return rows;
+    }
+
+    public Position getLeaf(int userLocation) {
+        Position userPosition = new Position(userLocation);
+        for (Row row : rows) {
+            int movePoint = row.getMovePoint(userPosition.getCurrentPosition());
```

Comment:
```suggestion
            int movePoint = row.getMovePoint(userPosition);
```

객체 자체를 넘겨주는건 어떨까요?

```
public class Row {

    private final Map<...> connectionTypes;

```

결국 포지션을 통해 무언가 찾는거라면, `Position` 자체가 키가 되어도 될것 같다는 생각이 드네요.
그렇다면 객체에게 값을 가지고 오지 않아도 되겠네요 🤔 

--- 

> 사실 객체지향이라는게 개인적으로는 비효율적인 부분이 많다고 생각해요. 로직으로 보나 메모리 관리로 보나.
이런관점도 있을수 있구나 정도로 봐주세요~